ruby/ql/lib/codeql/files/FilesImpl.qll

```diff
@@ -0,0 +1,21 @@
+private import codeql.utils.FileSystem as FS
+
+module FilesImpl implements FS::FilesSig {
+  class AtFile extends @file, AtContainer {
```

Comment:
I see you changed `FileSig` to define proper classes instead of a bunch of predicates that are to be aliased to the matching database relations and classes that match directly to database `@types`. 

Not sure which way is preferred. Your approach introduces an additional bunch of classes with the same hierarchy as in `Make`. Not sure if the optimizer can get rid of that. I think the more direct approach should result in the exact same code as the old-style hand-written implementations when the parametrised modules are instantiated.

Reply:
I think the optimizer will get rid of it once we reach the RA level, but I'm not too sure.   
I'll do DCA experiments once I got a polished implementation and the compiler crash that's currently blocking me is fixed. 

I like my approach (but maybe I should do something about the names). 
You had both a `files_` and `folders_` predicate in your signature, which I've collapsed into a single `getAbsolutePath()` in `AtContainer.  
You like chose to make those two predicate because the dbscheme has a `files()` and `folders()` relation, but that feels like an implementation detail.